test/test.oauth2.ts

```diff
@@ -1384,73 +1393,79 @@ describe('oauth2', () => {
       const expectedMetadata = {
         Authorization: 'Bearer access_token',
       };
+
       const requestMetaData = await client.getRequestHeaders(
         'http://example.com'
       );
+
       assert.deepStrictEqual(requestMetaData, expectedMetadata);
     });
 
-    it('should throw if tries to refresh but no refresh token is available', async () => {
+    it('should return authorization header on current access token if it has not expired', async () => {
```

Comment:
nit: `should return cached authorization header on getRequestHeaders() if not expired`